charts/kube-prometheus-stack/templates/grafana/configmaps-datasources.yaml

```
@@ -45,7 +45,7 @@ data:
 {{- end }}
 {{- if .Values.grafana.sidecar.datasources.createPrometheusReplicasDatasources }}
 {{- range until (int .Values.prometheus.prometheusSpec.replicas) }}
-    - name: {{ .Values.grafana.sidecar.datasources.name }}-{{ . }}
+    - name: "{{ .Values.grafana.sidecar.datasources.name }}-{{ . }}"
```

```suggestion
    - name: "{{ $.Values.grafana.sidecar.datasources.name }}-{{ . }}"
```

I guess, thats the real issue here. Since we are inside a range loop. `.Values` wont work anymore. like line 50 here.

---

Good catch 👍🏻